reactDashboard/src/components/DataCharts/DataCharts.jsx

```diff
@@ -278,6 +278,7 @@ const DataCharts = ({
       const chartConfiguration = commonChartConfiguration(
         datasets,
         darkmode,
+        maxDateRange,
```

Comment:
We discussed calculating the `maxDateRange` inside `commonChartConfiguration()` directly from the `datasets` parameter. 

- Pros are less parameters need be passed around, and there is less chance of the `datasets` and `maxDateRange` getting out of sync. 
- Cons are it would be recalculated every time `commonChartConfiguration()` was called. 

Opinions people?